public/app/features/correlations/types.ts

```diff
@@ -27,12 +27,18 @@ export interface RemoveCorrelationResponse {
 }
 
 type CorrelationTypeOptions = {
-  value: string;
+  value: CorrelationType;
   label: string;
   description: string;
 };
 
-export const CORR_TYPES: Record<string, CorrelationTypeOptions> = {
+export const CORR_TYPES_QUERY = {
+  value: 'query' as 'query',
+  label: 'Query',
+  description: 'Open a query',
+};
+
+export const CORR_TYPES: Record<CorrelationType, CorrelationTypeOptions> = {
```

Comment:
This won't work well for discriminating the union because TS does not actually know the values and just sees that `value: string`. I think using literal strings in checks like `correlation.type === 'query'` is fine because you get autocomplete and an error if you test with something else than `query` or `external`. If we still want to use a structure like this though we have to make TS aware that type of `value` is literally `query`. The `CORR_TYPES_QUERY` does that but with type assertion which will trip betterer and lint so other option would be to:
```
export const CORR_TYPES_QUERY: {value: 'query' ....} = {
  value: 'query',
  label: 'Query',
  description: 'Open a query',
};
```
but at that point not sure if that is worth it.